packages/valory/skills/registration_abci/behaviours.py

```diff
@@ -333,7 +337,36 @@ def request_update(self) -> Generator[None, None, bool]:
         self.updated_genesis_data.update(genesis_data)
         return True
 
-    def async_act(self) -> Generator:
+    def wait_for_block(self, timeout: float) -> Generator[None, None, bool]:
+        """Wait for a block to be received in the specified timeout."""
+        # every agent will finish with the reset at a different time
+        # hence the following will be different for all agents
+        start_time = datetime.datetime.now()
+
+        def received_block() -> bool:
+            """Check whether we have received a block after "start_time"."""
+            try:
+                shared_state = cast(SharedState, self.context.state)
+                last_timestamp = shared_state.round_sequence.last_timestamp
+                if last_timestamp > start_time:
```

Comment:
Would that make sense? We might have already received a block.

```suggestion
        start_time =  shared_state.round_sequence.last_timestamp or datetime.datetime.now()

        def received_block() -> bool:
            """Check whether we have received a block after "start_time"."""
            try:
                shared_state = cast(SharedState, self.context.state)
                last_timestamp = shared_state.round_sequence.last_timestamp
                if last_timestamp:
```

Reply:
We need to know that its a block received after the reset, the block *CAN* be before the reset as well.

Reply:
Ok so we need the comparison! How about the initialization? To avoid the `datetime.datetime.now()` if possible. Even better if we could wait for the first block before starting so that we do not use local time at all.

Reply:
So this behaviour is for agents to share configs with each other and learn about their tm configs. Before this behaviour is finished the tm nodes are not connected to each other, in the e2e tests they are connected, but this will not be the case when this feature is used in reality. So what CAN happen is we receive a block before the reset, but that can be from the badly configured (unconnected) tendermint node. We want blocks from the properly connected tm node.